sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/events/EventActionsTableSteps.java

```diff
@@ -0,0 +1,190 @@
+import javax.inject.Inject;
+import org.apache.commons.lang3.StringUtils;
+import org.assertj.core.api.SoftAssertions;
+import org.openqa.selenium.WebElement;
+import org.sormas.e2etests.helpers.WebDriverHelpers;
+import org.sormas.e2etests.pojo.web.EventActionTableEntry;
+import org.sormas.e2etests.state.ApiState;
+import org.sormas.e2etests.steps.BaseSteps;
+
+public class EventActionsTableSteps implements En {
+
+  private final WebDriverHelpers webDriverHelpers;
+  private final BaseSteps baseSteps;
+  private EventActionTableEntry eventActionTableEntry;
+
+  @Inject
+  public EventActionsTableSteps(
+      WebDriverHelpers webDriverHelpers,
+      BaseSteps baseSteps,
+      ApiState apiState,
+      SoftAssertions softly) {
+    this.webDriverHelpers = webDriverHelpers;
+    this.baseSteps = baseSteps;
+
+    Then(
+        "I search last created Event by API using EVENT UUID and wait for (\\d+) entries in the table",
+        (Integer expectedEntries) -> {
+          // webDriverHelpers.waitUntilElementIsVisibleAndClickable(FILTER_BY_GENERAL_INPUT);
+          webDriverHelpers.fillAndSubmitInWebElement(
+              FILTER_BY_GENERAL_INPUT,
+              getPartialUuidFromAssociatedLink(apiState.getCreatedEvent().getUuid()));
+          webDriverHelpers.waitUntilNumberOfElementsIsExactly(
+              EVENT_ACTIONS_TABLE_ROW, expectedEntries);
+        });
+
+    Then(
+        "^I am checking if all the fields are correctly displayed in the Event directory Actions table$",
+        () -> {
+          softly
+              .assertThat(eventActionTableEntry.getEventId())
+              .containsIgnoringCase(
+                  getPartialUuidFromAssociatedLink(apiState.getCreatedEvent().getUuid()));
+          softly
+              .assertThat(eventActionTableEntry.getActionTitle())
+              .containsIgnoringCase(apiState.getCreatedAction().getTitle());
+          softly
+              .assertThat(eventActionTableEntry.getActionCreationDate().toString())
+              .containsIgnoringCase(apiState.getCreatedAction().getDate().toString());
+          softly
+              .assertThat(eventActionTableEntry.getActionChangeDate().toString())
+              .containsIgnoringCase(apiState.getCreatedAction().getDate().toString());
+
+          softly
+              .assertThat(eventActionTableEntry.getActionStatus())
+              .containsIgnoringCase(apiState.getCreatedAction().getActionStatus());
+
+          softly
+              .assertThat(eventActionTableEntry.getActionPriority())
+              .containsIgnoringCase(apiState.getCreatedAction().getPriority());
+
+          softly
+              .assertThat(eventActionTableEntry.getActionLastModifiedBy())
+              .containsIgnoringCase("National USER");
+          softly.assertAll();
+        });
+
+    Then(
+        "^I collect the event actions from table view$",
+        () -> {
+          List<Map<String, String>> tableRowsData = getTableRowsData();
```

Comment:
isn't better to check specific column data in each row? if something changes in the UI in the data rows order the test will fail

if yes, please take a look to the WebDriverHelpers. getValueFromTableRowUsingTheHeader(String headerValue, int rowIndex)

Reply:
In this case, following the filtering, there is only one data row in the table. 
But for the general use, I am think, rows once returned, should further queried based on some other unique value within the row, index will be always unreliable. And this can ony be done for each table, specifically.